invenio_rdm_records/services/communities/service.py

```diff
@@ -205,10 +203,20 @@ def _remove(self, identity, community_id, record):
         if community_id not in record.parent.communities.ids:
             raise RecordCommunityMissing(record.id, community_id)
 
-        # check permission here, per community: curator cannot remove another community
-        self.require_permission(
-            identity, "remove_community", record=record, community_id=community_id
-        )
+        try:
+            self.require_permission(
+                identity, "remove_community", record=record, community_id=community_id
+            )
+            # By default, admin/superuser has permission to do everything, so PermissionDeniedError won't be raised for admin in any case
+        except PermissionDeniedError as exc:
+            # If user doesn't have permission, then check if community is required
+            if current_app.config["RDM_COMMUNITY_REQUIRED_TO_PUBLISH"]:
+                # If community is required for a record and if it is the last community to remove, raise error
+                if len(record.parent.communities.ids) == 1:
+                    raise CannotRemoveCommunityError()
```

Comment:
for readability
```suggestion
            # if permission is denied, decide which error to raise based on config
            community_required = current_app.config["RDM_COMMUNITY_REQUIRED_TO_PUBLISH"]
            community_missing = len(record.parent.communities.ids) <= 1
            if community_required && community_missing:
                raise CannotRemoveCommunityError()
```

also, I wouldn't check `==1` because you might have zero communities in edge cases. fe.
when someone with permissions has published before the record, then user without permission goes to publish again, this if should also catch this case